consensus/src/block_storage/block_store.rs

```diff
@@ -551,6 +551,15 @@ impl BlockStore {
             Duration::ZERO
         }
     }
+
+    pub async fn wait_for_payload(&self, block: &Block) -> anyhow::Result<()> {
+        self.payload_manager.get_transactions(block).await?;
```

Comment:
is there a timeout inside?

Reply:
added a timeout here for now. plan to refactor for heuristics later.